search-provider/eks-search-app.c

```diff
@@ -185,17 +186,16 @@ dispatch_subtree (EksSubtreeDispatcher *dispatcher,
       g_hash_table_insert (info.cache, g_strdup (subnode), provider);
     }
 
-  GDBusInterfaceSkeleton *skeleton;
-  g_object_get (provider, "skeleton", &skeleton, NULL);
-  return skeleton;
+  return eks_provider_skeleton_for_interface (provider, interface);
 }
 
 static GPtrArray *
 eks_search_app_node_interface_infos ()
 {
-  GPtrArray *ptr_array = g_ptr_array_new_full (2, (GDestroyNotify) g_dbus_interface_info_unref);
+  GPtrArray *ptr_array = g_ptr_array_new_full (3, (GDestroyNotify) g_dbus_interface_info_unref);
```

Comment:
Just use `g_ptr_array_new_with_free_func()` here... The overhead to resize the array for such a small one is really negligible.